cura/MachineActionManager.py

```diff
@@ -38,11 +38,10 @@ def initialize(self):
         # Ensure that all containers that were registered before creation of this registry are also handled.
         # This should not have any effect, but it makes it safer if we ever refactor the order of things.
         for container in container_registry.findDefinitionContainers():
-            self._onContainerAdded(container)
+            self.addActionsForContainer(container)
 
-        container_registry.containerAdded.connect(self._onContainerAdded)
```

Comment:
Huh, I would expect this to... No longer register the machine actions upon adding a new printer during the runtime of Cura. I guess a test will point this out when I'm running the code...

Reply:
I figured out why it still works. In `MachineManager.addMachine`, it directly calls this function upon adding a new printer. I think that you removed the wrong line then of the two.

The solution with the `containerAdded` signal is better because it doesn't add any new dependencies. The solution with `MachineManager.addMachine` calling `addActionsForContainer` adds a dependency of `MachineManager` to `MachineActionManager` which is illogical and circular.